src/components/mfa/MultiFactorAuthentication.tsx

```diff
@@ -0,0 +1,244 @@
+import * as React from 'react'
+import {
+  Box,
+  Button,
+  Chip,
+  CircularProgress,
+  Divider,
+  Grid,
+  Icon,
+  Paper,
+  Tooltip,
+  Typography,
+  styled,
+} from '@mui/material'
+import { LoadingButton } from '@mui/lab'
+import ConfirmDialog from '../ConfirmDialog'
+import MfaDialog from './MfaDialog'
+import ErrorSnackbar from '../ErrorSnackbar'
+import MaterialIcon from '../MaterialIcon'
+import useAppUserMfa from '../../hooks/useAppUserMfa'
+import ErrorMessage from '../messages/ErrorMessage'
+
+export const LargeIcon = styled(Icon)(({ theme }) => ({
+  fontSize: `${theme.typography.h4.fontSize} !important`,
+}))
+
+type Props = {
+  ssoSetupUrl?: string
+  isExternalIdentityProviderUser?: boolean
+}
+
+/**
+ * React Component that provides a mechanism for app users to configure Multi
+ * Factor Authentication. `<AppUserMfaProvider />` must be provided above this
+ * component in the component tree.
+ *
+ * #### Example
+ *
+ * ```js
+ * import * as React from 'react'
+ * import {
+ *   AppUserMfaProvider,
+ *   MultiFactorAuthentication,
+ * } from '@oneblink/apps-react'
+ *
+ * function Component() {
+ *   return <MultiFactorAuthentication />
+ * }
+ *
+ * function App() {
+ *   return (
+ *     <AppUserMfaProvider>
+ *       <Component />
+ *     </AppUserMfaProvider>
+ *   )
+ * }
+ *
+ * const root = document.getElementById('root')
+ * if (root) {
+ *   ReactDOM.render(<App />, root)
+ * }
+ * ```
+ *
+ * @param props
+ * @returns
+ * @group Components
+ */
+export default function MultiFactorAuthentication({
+  ssoSetupUrl,
+  isExternalIdentityProviderUser,
+}: Props) {
+  const {
+    loadingError,
+    setupError,
+    isLoading,
+    isMfaEnabled,
+    isDisablingMfa,
+    isSettingUpMfa,
+    mfaSetup,
+    loadAppUserMfa,
+    beginMfaSetup,
+    cancelMfaSetup,
+    completeMfaSetup,
+    clearMfaSetupError,
+    beginDisablingMfa,
+    completeDisablingMfa,
+    cancelDisablingMfa,
+  } = useAppUserMfa()
+
+  return (
+    <Grid item xs={true} lg={8}>
+      <Box padding={3}>
+        <Paper>
+          <Box padding={3}>
+            <Grid container spacing={2} alignItems="center">
+              <Grid item xs>
+                <Typography variant="h4" fontWeight="light">
+                  Multi Factor Authentication{' '}
+                  {isExternalIdentityProviderUser ? undefined : isLoading ? (
```

Comment:
Would be good if we could avoid the long ternaries, maybe rip this out into a new sub-component and render it like :

if(isExternalIdentityProviderUser) {
   return undefined
}
if(isLoading) {
  return loading component
}

etc